packages/graphql/src/schema/scalars/Duration.ts

```diff
@@ -0,0 +1,147 @@
+    } = match.groups!;
+
+    // NOTE: xUnit and xDT cannot both be nonzero by construction => (xUnit + xDT) = xUnit | xDT | 0
+    const years = +yearUnit + +yearDT;
+    const months = +monthUnit + +monthDT;
+    const weeks = +weekUnit;
+    const days = +dayUnit + +dayDT;
+    const hours = +hourUnit + +hourDT;
+    const minutes = +minuteUnit + +minuteDT;
+    const seconds = +secondUnit + +secondDT;
+
+    // Splits a component into a whole part and remainder
+    const splitComponent = (component: number): [number, number] => [
+        Math.trunc(component),
+        +(component % 1).toPrecision(9),
+    ];
+
+    // Calculate months based off of months and years
+    const [wholeMonths, remainderMonths] = splitComponent(months + years * MONTHS_PER_YEAR);
+
+    // Calculate days based off of days, weeks, and remainder of months
+    const [wholeDays, remainderDays] = splitComponent(days + weeks * DAYS_PER_WEEK + remainderMonths * DAYS_PER_MONTH);
+
+    // Calculate seconds based off of remainder of days, hours, minutes, and seconds
+    const splitHoursInSeconds = splitComponent((hours + remainderDays * HOURS_PER_DAY) * SECONDS_PER_HOUR);
+    const splitMinutesInSeconds = splitComponent(minutes * SECONDS_PER_MINUTE);
+    const splitSeconds = splitComponent(seconds);
+    // Total seconds by adding splits of hour minute second
+    const [wholeSeconds, remainderSeconds] = splitHoursInSeconds.map(
+        (p, i) => p + splitMinutesInSeconds[i] + splitSeconds[i]
+    );
+
+    // Calculate nanoseconds based off of remainder of seconds
+    const wholeNanoseconds = +remainderSeconds.toFixed(9) * 1000000000;
+
+    // Whether total duration is negative
+    const coefficient = negated ? -1 : 1;
+    // coefficient of duration and % may negate zero: converts -0 -> 0
+    const unsignZero = (a: number) => (Object.is(a, -0) ? 0 : a);
+
+    return {
+        months: unsignZero(coefficient * wholeMonths),
+        days: unsignZero(coefficient * wholeDays),
+        seconds: unsignZero(coefficient * wholeSeconds),
+        nanoseconds: unsignZero(coefficient * wholeNanoseconds),
+    };
+};
+
+const parse = (value: any) => {
+    const { months, days, seconds, nanoseconds } = parseDuration(value);
+
+    return new neo4j.types.Duration(months, days, seconds, nanoseconds);
+};
+
+export default new GraphQLScalarType({
+    name: "Duration",
+    description: "A duration, represented as an ISO 8601 duration string",
+    serialize: (value: any) => {
+        if (typeof value !== "string" && !(value instanceof neo4j.types.Duration)) {
+            throw new TypeError(`Value must be of type string: ${value}`);
+        }
+
+        const stringifiedValue = value.toString();
+
+        if (!DURATION_REGEX.test(stringifiedValue)) {
+            throw new TypeError(`Value must be formatted as Duration: ${stringifiedValue}`);
+        }
+
+        return stringifiedValue;
+    },
+    parseValue: (value) => {
```

Comment:
`value` is always a `string` in `parseValue`, so you can stipulate that here and get rid of the first type check in `parseDuration`.

Reply:
It's JSON correct? It makes sense that this is a string, but I was relying of `graphql` which types this as `any`. I thought I was missing something.

Reply:
Oh damn, looking back, maybe I'm the one who's missing something! 🙈  Honestly, I was basing it off the code in this project, for example the `BigInt` scalar: https://github.com/neo4j/graphql/blob/master/packages/graphql/src/schema/scalars/BigInt.ts#L30

But delving into it, perhaps it should be typed as `any`, my mistake. 😞  First day back after a week off yesterday, can I be excused? 😄 

Definitely remove the type check as you have done in `parseDuration`, but you're right, it probably does need to be in here.

Reply:
No worries. Added back in https://github.com/neo4j/graphql/pull/459/commits/0d7f83fb29a8d08796a551ee070bb1dcecace78a